mdata/aggmetric.go

```diff
@@ -80,16 +80,21 @@ func NewAggMetric(store Store, cachePusher cache.CachePusher, key schema.AMKey,
 }
 
 // Sync the saved state of a chunk by its T0.
-func (a *AggMetric) SyncChunkSaveState(ts uint32) {
-	a.Lock()
-	defer a.Unlock()
-	if ts > a.lastSaveFinish {
-		a.lastSaveFinish = ts
-	}
-	if ts > a.lastSaveStart {
-		a.lastSaveStart = ts
+func (a *AggMetric) SyncChunkSaveState(ts uint32, sendPersist bool) ChunkSaveCallback {
+	return func() {
+		a.Lock()
+		defer a.Unlock()
```

Comment:
we need to release the lock before calling SendPersistMessage().

KafkaNotifier  uses an unbuffered channel for queuing PersistMessages so we could block here (probably not for long) waiting for other persistMessages to be handled.